openmc/deplete/abc.py

```diff
@@ -364,6 +365,219 @@ def nuclides(self, nuclides):
+    """
+
+    def __init__(self, chain_nuclides):
+        self._chain_nuclides = {}
+        self._constant_yields = {}
+
+        # Get all nuclides with fission yield data
+        for nuc in chain_nuclides:
+            if len(nuc.yield_data) == 1:
+                self._constant_yields[nuc.name] = (
+                    nuc.yield_data[nuc.yield_energies[0]])
+            elif len(nuc.yield_data) > 1:
+                self._chain_nuclides[nuc.name] = nuc
+        self._chain_set = set(self._chain_nuclides) | set(self._constant_yields)
+
+    @property
+    def constant_yields(self):
+        out = {}
+        for key, sub in self._constant_yields.items():
+            out[key] = sub.copy()
+        return out
+
+    @abstractmethod
+    def weighted_yields(self, local_mat_index):
+        """Return fission yields for a specific material
+
+        Parameters
+        ----------
+        local_mat_index : int
+            Index for material tracked on this process that
+            exists in :attr:`local_mat_index` and fits within
+            the first axis in :attr:`results`
+
+        Returns
+        -------
+        library : dict
+            Dictionary of ``{parent: {product: fyield}}``
+        """
+
+    @staticmethod
+    def unpack():
+        """Unpack tally data prior to compute fission yields.
+
+        Called after a :meth:`openmc.deplete.Operator.__call__`
+        routine during the normalization of reaction rates.
+
+        Not necessary for all subclasses to implement, unless tallies
+        are used.
+        """
+
+    @staticmethod
+    def generate_tallies(materials, mat_indexes):
+        """Construct tallies necessary for computing fission yields
+
+        Called during the operator set up phase prior to depleting.
+        Not necessary for subclasses to implement
+
+        Parameters
+        ----------
+        materials : iterable of C-API materials
+            Materials to be used in :class:`openmc.capi.MaterialFilter`
+        mat_indexes : iterable of int
+            Indices of tallied materials that will have their fission
+            yields computed by this helper. Necessary as the
+            :class:`openmc.deplete.Operator` that uses this helper
+            may only burn a subset of all materials when running
+            in parallel mode.
+        """
+
+    def update_tally_nuclides(self, nuclides):
+        """Return nuclides with non-zero densities and yield data
+
+        Parameters
+        ----------
+        nuclides : iterable of str
+            Nuclides with non-zero densities from the
+            :class:`openmc.deplete.Operator`
+
+        Returns
+        -------
+        nuclides : tuple of str
+            Union of nuclides that the :class:`openmc.deplete.Operator`
+            says have non-zero densities at this stage and those that
+            have yield data. Sorted by nuclide name
+
+        """
+        return tuple(sorted(self._chain_set & set(nuclides)))
```

Comment:
Personal preference? These are intended to be used in the `Tally.nuclides` setter, so a list would also be sufficient. In my experience, tuples are a hair faster for iteration. We also don't need mutability on these. But compared to time and memory taken up by transport, this will be largely inconsequential

Reply:
Returned as lists